nixos/modules/security/please.nix

```diff
@@ -0,0 +1,106 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.security.please;
+  ini = pkgs.formats.ini { };
+in
+{
+  options.security.please = {
+    enable = mkEnableOption (mdDoc ''
+      please, a Sudo clone which allows a users to execute a command or edit a
+      file as another user
+    '');
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.please;
+      defaultText = literalExpression "pkgs.please";
+      description = mdDoc ''
+        Which package to use for {command}`please`;
+      '';
+    };
+
+    wheelNeedsPassword = mkOption {
+      type = types.bool;
+      default = true;
+      description = lib.mdDoc ''
+        Whether users of the `wheel` group must provide a password to run
+        commands or edit files with {command}`please` and
+        {command}`pleaseedit` respectively.
+      '';
+    };
+
+    settings = mkOption {
+      type = ini.type;
+      default = { };
+      example = {
+        jim_become_root = {
+          name = "jim";
+          target = "root";
+          rule = ".*";
+          require_pass = false;
+        };
+        jim_hosts = {
+          name = "jim";
+          type = "edit";
+          target = "root";
+          rule = "/etc/hosts";
+          editmode = 644;
+          require_pass = false;
+        };
+      };
+      description = mdDoc ''
+        Please configuration. Refer to
+        <https://github.com/edneville/please/blob/master/please.ini.md> for
+        details.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    security.wrappers =
+      let
+        owner = "root";
+        group = "root";
+        setuid = true;
+      in
+      {
+        please = {
+          source = "${cfg.package}/bin/please";
+          inherit owner group setuid;
+        };
+        pleaseedit = {
+          source = "${cfg.package}/bin/pleaseedit";
+          inherit owner group setuid;
+        };
+      };
+
+    security.please.settings = rec {
+      wheel_run_as_root = {
+        type = "run";
+        name = "wheel";
+        group = true;
+        target = "root";
```

Comment:
I wonder if this should become `.*`, since `wheel` should also be able to become any user (just like with `sudo` and `doas`). (If you decide to do this, I'd also change the names to be something like `wheel_run` or `wheel_run_as_any`.)

Reply:
Now `root` can do anything by default. I couldn't make `mkForce` or something similar to work for the INI generator, so I just appended this configuration to the exposed option via `//`: this will ignore attributes with the same name on generation and won't error out (which I don't think is ideal, but I can't find other solutions to this).

Reply:
Makes sense. Changed.